product_configurator_wizard/wizard/product_configurator.py

```diff
@@ -805,14 +805,17 @@ def action_config_done(self):
         if self.product_id:
             remove_cv_links = map(lambda cv: (2, cv), self.product_id.value_custom_ids.ids)
             new_cv_links = self.product_id.product_tmpl_id.encode_custom_values(custom_vals)
-            self.product_id.write({
-                'attribute_value_ids': [(6, 0, self.value_ids.ids)],
-                'value_custom_ids':  remove_cv_links + new_cv_links,
-            })
-            if self.order_line_id:
-                self.order_line_id.write(self._extra_line_values(self.order_line_id.order_id,
-                                                                 self.product_id,
-                                                                 new=False))
+
+            if self.product_id.attribute_value_ids.ids != self.value_ids.ids:
+                new_product = self.product_tmpl_id.create_variant(self.value_ids.ids, custom_vals)
+                self.order_line_id.write({'product_id': new_product.id})
```

Comment:
Here https://github.com/pledra/odoo-product-configurator/pull/30/files#diff-dc0d573fe7a7ffdcfaac0b70bd2d0ff7L812 you remove e method introduced by Richard (self._extra_line_values).

This hook method is used to update the sale.order.line object with the line's name and also gives the user the ability to add extra values to the write.

So instead of writing directly to the order line and passing only product_id as you do here: https://github.com/pledra/odoo-product-configurator/pull/30/files#diff-dc0d573fe7a7ffdcfaac0b70bd2d0ff7R811

Best is to call the method just before self.unlink() with the same if condition that was removed and you can also add the product_id by default in the method